dnf/logging.py

```diff
@@ -87,14 +89,37 @@ def _cfg_err_val2level(cfg_errval):
     assert 0 <= cfg_errval <= 10
     return _ERR_VAL_MAPPING.get(cfg_errval, logging.WARNING)
 
-def _create_filehandler(logfile):
+
+class MultiprocessRotatingFileHandler(logging.handlers.RotatingFileHandler):
+    def __init__(self, filename, mode='a', maxBytes=0, backupCount=0, encoding=None, delay=False):
+        logging.handlers.RotatingFileHandler.__init__(
+            self, filename, mode, maxBytes, backupCount, encoding, delay,
+        )
+        self.rotate_lock = dnf.lock.build_log_lock("/var/log/", True)
+
+    def emit(self, record):
+        while True:
+            try:
+                if self.shouldRollover(record):
+                    with self.rotate_lock:
+                        self.doRollover()
+                logging.FileHandler.emit(self, record)
+                return
+            except (dnf.exceptions.ProcessLockError, dnf.exceptions.ThreadLockError):
+                pass
```

Comment:
shouldn't something like 'time.sleep(1)' be here?

Reply:
I added time.sleep(0.01). Not sure if this is the best value, but under normal circumstances, the exception almost never occurs, so it doesn't matter all that much.